helpers/refreshAchievements.ts

```diff
@@ -3,27 +3,72 @@ import AchievementType from '@root/constants/achievements/achievementType';
 import DiscordChannel from '@root/constants/discordChannel';
 import { GameId } from '@root/constants/GameId';
 import { Games } from '@root/constants/Games';
+import Role from '@root/constants/role';
 import { getSolvesByDifficultyTable } from '@root/helpers/getSolvesByDifficultyTable';
 import { createNewAchievement } from '@root/helpers/notificationHelper';
 import { getDifficultyRollingSum } from '@root/helpers/playerRankHelper';
 import Achievement from '@root/models/db/achievement';
+import Comment from '@root/models/db/comment';
 import Level from '@root/models/db/level';
 import MultiplayerMatch from '@root/models/db/multiplayerMatch';
 import MultiplayerProfile from '@root/models/db/multiplayerProfile';
 import Review from '@root/models/db/review';
 import User from '@root/models/db/user';
-import { AchievementModel, LevelModel, MultiplayerMatchModel, MultiplayerProfileModel, ReviewModel, UserConfigModel, UserModel } from '@root/models/mongoose';
+import { AchievementModel, CommentModel, LevelModel, MultiplayerMatchModel, MultiplayerProfileModel, ReviewModel, UserConfigModel, UserModel } from '@root/models/mongoose';
 import { Types } from 'mongoose';
 import queueDiscordWebhook from './discordWebhook';
 import { getRecordsByUserId } from './getRecordsByUserId';
 
 const AchievementCategoryFetch = {
+  [AchievementCategory.SOCIAL]: async (_gameId: GameId, userId: Types.ObjectId) => {
+    // no game ID as this is a global
+    const [commentCount, welcomedComments] = await Promise.all([
+      CommentModel.countDocuments({
+        author: userId,
+        deletedAt: null,
+        target: { $ne: userId }
+      }),
+      CommentModel.find({
+        author: userId,
+        deletedAt: null,
+        target: { $ne: userId },
+        text: { $regex: /welcome/i },
```

Comment:
not sure about this, maybe we should just reward comments made within the first 24hrs instead of requiring them to say "welcome"?